datanode/gateway/graphql/oracles_resolver.go

```diff
@@ -67,25 +69,112 @@ func (o *oracleSpecResolver) DataConnection(ctx context.Context, obj *vegapb.Ora
 type oracleDataResolver VegaResolverRoot
 
 func (o *oracleDataResolver) ExternalData(ctx context.Context, obj *vegapb.OracleData) (*ExternalData, error) {
+	ed := &ExternalData{
+		Data: &Data{},
+	}
+
 	if obj != nil {
-		var signers []*Signer
-		if len(obj.ExternalData.Data.Signers) > 0 {
-			signers = make([]*Signer, len(obj.ExternalData.Data.Signers))
-			for i, signer := range obj.ExternalData.Data.Signers {
-				signers[i] = &Signer{
-					Signer: signer.GetSigner().(SignerKind),
+		if obj.ExternalData != nil && obj.ExternalData.Data != nil {
+			o := obj.ExternalData.Data
+			broadcastAt := strconv.FormatInt(o.BroadcastAt, 10)
+			matchedSpecs := []string{}
+			if len(o.MatchedSpecIds) > 0 {
+				matchedSpecs = o.MatchedSpecIds
+			}
```

Comment:
These are the same types. Unless you wanted to ensure that we don't get `null` in the response, but instead at least an empty string array, in which case you could do:
```suggestion
			if len(o.MatchedSpecIds) == 0 {
				o.MatchedSpecIds = []string{}
			}
```

Reply:
Yes, corrected that here https://github.com/vegaprotocol/vega/pull/6750/commits/439e167fbb5953adcc6b6d7a481ea70596b19512#diff-0cde61576df49d23ac0103aca13202a605d1c83edf31d12fdc3d5281c83005c9R79-R80
Thanks!